src/operator/nn/mkldnn/mkldnn_act.cc

```diff
@@ -107,36 +107,46 @@ class MKLDNNActForward {
                    const NDArray &data, const mkldnn::memory &mem): fwd_pd(
                        GetActFwdDescImpl(param, is_train, mem, data.dtype())) {
   }
+  void SetDataHandle(const NDArray &in_data, const NDArray &out_data);
+  void Execute();
+};  // class MKLDNNActForward
 
-  void SetNewMem(const mkldnn::memory &data, const mkldnn::memory &output) {
-    if (this->data == nullptr)
+void MKLDNNActForward::SetDataHandle(const NDArray &in_data, const NDArray &out_data) {
+    auto input_mem = in_data.GetMKLDNNData();
+    auto out_mem = const_cast<NDArray &>(out_data).CreateMKLDNNData(
+        fwd_pd.dst_primitive_desc());
+
+    if (this->data == nullptr) {
```

Comment:
I think it would be better to add initialization and first creation in the constructive function the MKLDNNActForward class. So there is no need to chech nullptr for every iteration.
Besides, please follow the coding style and naming methods we discussed in previous email.